### PR TITLE
feat(task): refactor task getStatus interface return TaskStatus instead of JobStatus

### PR DESCRIPTION
**Pull request by LioRoger**
#### What type of PR is this?

type-feature

#### What this PR does / why we need it:
let task get status return TaskStatus, That means task will not known nothing  about job.  job status will be refreshed depends on task status

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/state/JobStatusFsm.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     */
+    public JobStatus determinateJobStatus(JobStatus currentStatus, TaskStatus taskStatus) {
+        switch (taskStatus) {
+            // prepare is task init status, job should expected in running status
+            case PREPARING:
+                // running meaning task has running normally, job should expected in running status
+            case RUNNING:
+                // if task status is canceled, it must be canceled by job who owns it
+            case CANCELED:
+                return currentStatus;
```
Comment (guowl3) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/state/JobStatusFsm.java:43]: could there be an illegal status here?

Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/state/JobStatusFsm.java:43]: 1.taskstatus = prepareing ==> job = running is expected, if job = canceling,  outside invoke timeout -> cancel
2.taskstatus = cancel ==> job -> cancel
3.taskStatus = running ==> job -> running remain, job -> canceling, remain,

Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/state/JobStatusFsm.java:43]: 1.taskstatus = prepareing ==> job = running is expected, if job = canceling, outside invoke timeout -> cancel
2.taskstatus = cancel ==> job -> cancel
3.taskStatus = running ==> job -> running remain, job -> canceling, remain,

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java [hunk 2]
```diff
@@ -442,30 +386,76 @@ private void doRefreshResult(Long id) throws JobException {
...
                 .execute(() -> publisher.publishEvent(new DefaultJobProcessUpdateEvent(result)));
 
         if (publisher != null && result.getStatus() != null && result.getStatus().isTerminated()) {
             taskResultPublisherExecutor.execute(() -> publisher
-                    .publishEvent(new JobTerminateEvent(result.getJobIdentity(), result.getStatus())));
+                    .publishEvent(new JobTerminateEvent(result.getJobIdentity(), expectedJobStatus)));
 
             // TODO maybe we can destroy the pod there.
-            if (result.getStatus() == JobStatus.FAILED) {
+            if (result.getStatus() == TaskStatus.FAILED) {
```
[the remark this answers is not in the document]
Comment (LioRoger) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java:409]: it will be moved in future commit

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request closed without merging**